### ISSUE TITLE
Separating hair into their own outfit slots makes it generate hair shaders improperly.

### ISSUE THREAD

**Issue by johnvo0918** [FlailingFog/KK-Blender-Porter-Pack#596]
I'm separating them in Koi so that I don't have to separate them later in blender and potentially messing up the bone hierarchy which messes up hair physics later on (for some reason this happens sometimes for me, but please help me confirm that it's not because of separating)

Separated hair into outfit slots(this is upon importing file stays untouched):
![image](https://github.com/user-attachments/assets/34362912-f20e-4c0e-8114-c69a6f5e9437)

Not separated and all in 1 slot (ignore the orange edited the shader):
![image](https://github.com/user-attachments/assets/ac62c984-9674-48cb-9729-09d78501a6ae)

Adding a top item helps fixing this problem but it doesn't work all the time.
This problem only occurs on Hair. if they're all in 1 outfit slot the shader will generate properly.

card (separated hair):
![Koikatu_F_20240903172128112_Kaiz Optimized_K](https://github.com/user-attachments/assets/602135ad-a544-4bb7-896c-95fa675284f2)
card (not separated hair):
![Koikatu_F_20240722163803473_Kaiz Optimized_K](https://github.com/user-attachments/assets/02aa2aa3-cfce-477e-a039-2379ebbed5f3)

**Comment by FlailingFog** [comment]
I don't know why that's happening, but it looks like the importer isn't setting up the outline modifier correctly. Turning it off makes the hair look right.
![image](https://github.com/user-attachments/assets/34b6397f-b579-41f6-bce7-841ff11650c2)

If you want to keep the outline enabled, you can play with the "material offset" number in the outline modifier options shown above. The importer should set this automatically but it looks like incorrectly it's counting the materials... 
For example, on hair outfit 11 it should count up to the outline materials in the material list (which is slot 14) and set the material offset to 14, but it's incorrectly setting it to 3...
![image](https://github.com/user-attachments/assets/5674cf0c-86ed-4a4c-9979-4b88893606a1)

**Comment by johnvo0918** [comment]
yeah i'm fine with not having the outlines. I'm more interested in putting the model into Unity to Warudo/VRChat than using it in blender. and unity shaders can handle outlines too!

**Comment by FlailingFog** [comment]
It's still happening on KKBP 8.0, but if you aren't using the outlines, I guess it doesn't matter
